### PR TITLE
Fix issue where device code was relocated too early in library builds.

### PR DESCRIPTION
**Pull request by robinson96**
Fixes an issue where LINK_WITH_NVCC would cause device symbols in libraries to be invisible at final link time. Only affected libraries that provided device side functionality, callable from kernels.

### CODE REVIEW

File: cmake/thirdparty/FindCUDA.cmake [hunk 1]
```diff
@@ -1844,6 +1830,8 @@ macro(CUDA_ADD_EXECUTABLE cuda_target)
   ####################
   if (NOT CUDA_LINK_WITH_NVCC)
       CUDA_COMPUTE_SEPARABLE_COMPILATION_OBJECT_FILE_NAME(link_file ${cuda_target} "${${cuda_target}_SEPARABLE_COMPILATION_OBJECTS}")
+  else()
+     set(link_file "")
   endif()
```
Comment (cyrush) [inline cmake/thirdparty/FindCUDA.cmake]: hard to tell on quick review -- these stmts are already enclosed in BLT CHANGE comment blocks?

File: cmake/thirdparty/FindCUDA.cmake [hunk 2]
```diff
@@ -1872,7 +1860,7 @@ macro(CUDA_ADD_EXECUTABLE cuda_target)
       # We need to link with nvcc. Host linking prevents calling device methods in one library from kernels in another (yes, even with separable
       # compilation)
       string (REPLACE ";" " " LD_FLAGS_STR "${CUDA_NVCC_FLAGS}")
-      set (CMAKE_CUDA_LINK_EXECUTABLE  "${CUDA_NVCC_EXECUTABLE}  <FLAGS> <LINK_FLAGS> <OBJECTS>  -o <TARGET> <LINK_LIBRARIES>" CACHE PATH "" )
+      set (CMAKE_CUDA_LINK_EXECUTABLE  "${CUDA_NVCC_EXECUTABLE}  <FLAGS> <LINK_FLAGS> <OBJECTS>  -o <TARGET> <LINK_LIBRARIES>" CACHE STRING "" )
       set_target_properties(${cuda_target} PROPERTIES LINKER_LANGUAGE "CUDA")
```
Comment (cyrush) [inline cmake/thirdparty/FindCUDA.cmake]: hard to tell on quick review -- these stmts are already enclosed in BLT CHANGE comment blocks?

File: cmake/thirdparty/FindCUDA.cmake (hunk 1 above)
Comment (robinson96) [reply cmake/thirdparty/FindCUDA.cmake]: yup

File: cmake/thirdparty/FindCUDA.cmake (hunk 2 above)
Comment (robinson96) [reply cmake/thirdparty/FindCUDA.cmake]: this one isn't. Chris and I talked about at and we thought the change was minor enough to not warrant it...

Comment (cyrush) [reply cmake/thirdparty/FindCUDA.cmake]: agreed

**Pull request merged**